tests/rados/test_osd_rebalance.py

```diff
@@ -28,68 +30,87 @@ def run(ceph_cluster, **kw):
-            log.error("Failed to retrieve acting pg set")
-            return 1
-        osd_id = acting_pg_set[0]
-        host = rados_obj.fetch_host_node(daemon_type="osd", daemon_id=osd_id)
-        if not host:
-            log.error("Failed to fetch host details")
-            return 1
-        # fetch container id
-        out, _ = host.exec_command(sudo=True, cmd="podman ps --format json")
-        container_id = [
-            item["Names"][0]
-            for item in json.loads(out.read().decode())
-            if f"osd.{osd_id}" in item["Command"]
-        ][0]
-        if not container_id:
-            log.error("Failed to retrieve container id")
-            return 1
-        # fetch device path by osd_id
-        vol_out, _ = host.exec_command(
-            sudo=True,
-            cmd=f"podman exec {container_id} ceph-volume lvm list --format json",
-        )
-        volume_out = vol_out.read().decode()
-        dev_path = [
-            v[0]["devices"][0]
-            for k, v in json.loads(volume_out).items()
-            if str(k) == str(osd_id)
-        ][0]
-        if not dev_path:
-            log.error("Failed to get device path")
-            return 1
-        log.debug(
-            f"device path  : {dev_path}, osd_id : {osd_id}, host.hostname : {host.hostname}"
-        )
-        utils.set_osd_devices_unamanged(ceph_cluster, unmanaged=True)
-        method_should_succeed(rados_obj.bench_write, **pool)
-        method_should_succeed(utils.set_osd_out, ceph_cluster, osd_id)
-        method_should_succeed(wait_for_clean_pg_sets, rados_obj)
-        utils.osd_remove(ceph_cluster, osd_id)
-        method_should_succeed(wait_for_clean_pg_sets, rados_obj)
-        method_should_succeed(utils.zap_device, ceph_cluster, host.hostname, dev_path)
-        method_should_succeed(
-            wait_for_device, host, container_id, osd_id, action="remove"
-        )
-        utils.add_osd(ceph_cluster, host.hostname, dev_path, osd_id)
-        method_should_succeed(wait_for_device, host, container_id, osd_id, action="add")
-        method_should_succeed(wait_for_clean_pg_sets, rados_obj)
-        method_should_succeed(rados_obj.bench_write, **pool)
-        method_should_succeed(wait_for_clean_pg_sets, rados_obj)
-        utils.set_osd_devices_unamanged(ceph_cluster, unmanaged=False)
-        rados_obj.change_recover_threads(config=pool, action="rm")
+    if config.get("create_pools"):
+        pools = config.get("create_pools")
+        for each_pool in pools:
+            cr_pool = each_pool["create_pool"]
+            if cr_pool.get("pool_type", "replicated") == "erasure":
+                method_should_succeed(
+                    rados_obj.create_erasure_pool, name=cr_pool["pool_name"], **cr_pool
+                )
+            else:
+                method_should_succeed(
+                    rados_obj.create_pool,
+                    pool_name=cr_pool["pool_name"],
+                    pg_num=cr_pool["pg_num"],
```

Comment:
We need to send ` **kwargs ` here for attributes like ` size ` , ` min_size `, autoscale mode to be set for replicated pools.

Reply:
done